src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignerInfo.cs

```diff
@@ -838,19 +773,14 @@ private static int FindAttributeIndexByOid(AttributeAsn[] attributes, Oid oid, i
             return -1;
         }
 
-        private static int FindAttributeValueIndexByEncodedData(ReadOnlyMemory<byte> attributeValues, ReadOnlySpan<byte> asnEncodedData, out bool isOnlyValue)
+        private static int FindAttributeValueIndexByEncodedData(ReadOnlyMemory<byte>[] attributeValues, ReadOnlySpan<byte> asnEncodedData, out bool isOnlyValue)
```

Comment:
isn't this adding allocations?

Reply:
I could be wrong, but overall it should not make the situation worse than it is. Previously there was one big array holding all the values, which needed to be rebuilt from scratch on every change. Now there's basically array of arrays, which is more objects, but they are smaller and sometimes they reuse the data.

Reply:
I think 200 lines less is worth it but wondering if we can keep allocations the same and have less code